src/utils.js

```diff
@@ -1,37 +1,51 @@
 const BigNumber = require('bignumber.js');
+const RollingMath = require('rolling-math');
 const { ethers } = require('forta-agent');
 
-// create a fake function name
-function getRandomCharacterString(numCharacters) {
-  let result = '';
-  let charCode;
-  for (let i = 0; i < numCharacters; i++) {
-    charCode = Math.floor(Math.random() * 52);
-    if (charCode < 26) {
-      charCode += 65;
-    } else {
-      charCode += 97 - 26;
-    }
-    result += String.fromCharCode(charCode);
-  }
-  return result;
+function getAbi(agentName, abiFile) {
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const abiPath = `../abi/${agentName}`;
+  console.log(`getting: ${abiPath}/${abiFile}`);
+  const { abi } = require(`${abiPath}/${abiFile}`);
+  return abi;
 }
 
-function isAddress(valueString) {
-  return ethers.utils.isHexString(valueString, 20);
+function getInternalAbi(agentType, abiFile) {
+  // eslint-disable-next-line global-require,import/no-dynamic-require
+  const abiPath = `./${agentType}/internal-abi`;
+  console.log(`getting: ${abiPath}/${abiFile}`);
+  const { abi } = require(`${abiPath}/${abiFile}`);
+  return abi;
+}
+
+// helper function that identifies key strings in the args array obtained from log parsing
+// these key-value pairs will be added to the metadata as event args
+// all values are converted to strings so that BigNumbers are readable
+function extractEventArgs(args) {
+  const eventArgs = {};
+  Object.keys(args).forEach((key) => {
+    if (Number.isNaN(Number(key))) {
+      eventArgs[key] = args[key].toString();
+    }
+  });
+  return eventArgs;
 }
 
 function isNumeric(valueString) {
   const result = new BigNumber(valueString);
   return !(result.isNaN());
 }
 
+function isAddress(valueString) {
+  return ethers.utils.isHexString(valueString, 20);
+}
+
 function addressComparison(variable, operator, operand) {
   switch (operator) {
     case '===':
-      return variable.toLowerCase() === operand.toLowerCase();
+      return variable === operand;
     case '!==':
-      return variable.toLowerCase() !== operand.toLowerCase();
+      return variable !== operand;
```

Comment:
Unfortunately, these `.toLowerCase()` methods need to be present.  The address returned from any `ethers.js` method will be checksummed (mixed case) while addresses returned from the `forta-agent` methods will be lowercase.  To ensure that this function works, we are intentionally converting all addresses to lowercase.

We have discussed going in the opposite direction, namely using an `ethers.js` method (getAddress, I think), to convert all addresses to checksummed format, but we have not implemented that change (it would require going back through a lot of our repositories to enforce that change).